source/darwin/commoncrypto_aes.c

```diff
@@ -355,6 +353,37 @@ struct aws_symmetric_cipher *aws_aes_ctr_256_new_impl(
     return &cc_cipher->cipher_base;
 }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+/*
+ * Note that CCCryptorGCMFinal is deprecated in Mac 10.13. It also doesn't compare the tag with expected tag
+ * https://opensource.apple.com/source/CommonCrypto/CommonCrypto-60118.1.1/include/CommonCryptorSPI.h.auto.html
+ */
+static CCStatus s_cc_crypto_gcm_finalize(struct _CCCryptor *encryptor_handle, uint8_t *buffer, size_t tag_length) {
+#ifdef USE_LATEST_CRYPTO_API
+    if (__builtin_available(macOS 10.13, iOS 11.0, *)) {
```

Comment:
this makes sense, my only hold-back is the initial versioning macro at the top

Reply:
Discussed offline, and added more details in the PR description.